http-client/src/test/groovy/io/micronaut/http/client/filter/MutateRequestClientFilterSpec.groovy

```diff
@@ -55,7 +55,7 @@ class MutateRequestClientFilterSpec extends Specification {
 
     void "test mutate stream request URI"() {
         expect:
-        myClient.stream().blockFirst() == "xxxxxxxxxxx"
+        myClient.stream().collectList().block().join("") == '["xxxxxxxxxxx"]'
```

Comment:
bc String is a raw type, it doesn't deserialize it from array anymore. the controller already added the quotes manually so the test case was a bit weird in the first place

Reply:
to explain more:

The controller returns a Flux<String>, composed of a single string "xxx" including the quotes. The http stack does not serialize the string as json because it's a raw type (serializing it would yield double quoting "\"xxx\""). However it does add the brackets because it's not byte[] or ByteBuffer (yes it's weird, but this is the most compatible behavior). This hasn't changed.

On the client side, it used to split up the input and parse the string fully as json. This has changed, it treats it as a raw type because it's String, so it just yields the unparsed body now.